tests/functional/test_endpoints.py

```diff
@@ -171,3 +173,45 @@ def _assert_service_name_matches_endpoint_prefix(session, service_name):
         "Actual service name `%s` does not match expected service name "
         "we computed: `%s`" % (
             actual_service_name, computed_name))
+
+
+class TestEndpointResolution(BaseSessionTest):
```

Comment:
It might be worth if to see if we can move this over to `tests/functional/test_regions.py`. Mainly the tests there seem more specific to region/endpoint resolution. The tests in this file or more about making sure the information in our endpoints file matches up with the services we have.